src/main/java/frc/robot/subsystems/lilih/LilihSocket.java

```diff
@@ -17,35 +17,41 @@ public class LilihSocket {
 
   private WebsocketListener listener;
   private ObjectMapper objectMapper;
-  private boolean webSocketConnected;
-  int ip;
+  private int ip;
 
   public LilihSocket(int ip) {
-    HttpClient httpClient = HttpClient.newHttpClient();
-    listener = new WebsocketListener();
-    new Thread(
-            () -> {
-              try {
-                httpClient
-                    .newWebSocketBuilder()
-                    .buildAsync(URI.create("ws://10.43.29." + ip + ":5806"), listener)
-                    .get();
-                webSocketConnected = true;
-              } catch (InterruptedException | ExecutionException e) {
-                System.out.println(e.getMessage());
-              }
-            })
-        .start();
+    this.ip = ip;
+
+    createSocket();
+
     objectMapper = new ObjectMapper();
     objectMapper.configure(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES, false);
   }
 
+  private void createSocket() {
+    HttpClient httpClient = HttpClient.newHttpClient();
+    listener = new WebsocketListener();
+    try {
+      httpClient
+          .newWebSocketBuilder()
+          .buildAsync(URI.create("ws://10.43.29." + ip + ":5806"), listener)
+          .get();
+    } catch (InterruptedException | ExecutionException e) {
+      System.out.println(e.getMessage());
```

Comment:
```suggestion
      e.printStackTrace();
```
This'll get you not just the message on the exception, but the stack trace as well and help it stand out.